Userland/Libraries/LibWeb/HTML/EventLoop/TaskQueue.cpp

```diff
@@ -88,4 +88,13 @@ Task const* TaskQueue::last_added_task() const
     return m_tasks.last();
 }
 
+bool TaskQueue::has_rendering_tasks() const
+{
+    for (auto const& task : m_tasks) {
+        if (task->source() == Task::Source::Rendering)
+            return true;
+    }
+    return false;
```

Comment:
🤔 another reason to think about having separate vectors of tasks for different task sources I s'pose. A "single-task" task source might be useful. Keeping a total order might be hard in that case